### PR TITLE
feat: add OC.Files.Client.unlock() and add lock token to putFileContents

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description
OC.Files.Client now supports lock token in putFileContents() and unlock() was added

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by phil-davis** [comment]
@pako81 this has "3 - To Review" label, but nobody is mentioned in "Reviewers".
Also, there is no changelog.
Is this wanted for 10.13?

**Comment by pako81** [comment]
Not sure what is the context here. Still, maybe @jvillafanez could try to review this?

File: core/js/files/client.js [hunk 1]
```diff
@@ -800,6 +777,40 @@
 			return promise;
 		},
 
+		unlock: function(path, token, options) {
```
Comment (jvillafanez) [inline core/js/files/client.js:780]: Needs documentation, specially if it's expected to be used in different places.

**Comment by jvillafanez** [comment]
I haven't tested the code, but it looks good.

**Comment by pako81** [comment]
I guess @DeepDiver1975 is the only one we can address the requested change so probably this will not make into 10.13 or @jvillafanez do you have enough context for taking care of adding doc?

**Comment by jvillafanez** [comment]
I think I can add something in order to merge it, and then @DeepDiver1975 can make a different PR if there is something wrong

**Comment by pako81** [comment]
sounds great, thx

**Comment by pako81** [comment]
It looks like CI failed because of a temporary hiccup. It may be needed to restart it. We would also need to retarget this to the `release-10.13.0` branch if we want to have it in 10.13 as we already have code freeze.

**Comment by phil-davis** [comment]
The PR branch was from before the drone secrets changes, so it was running an old set of docker pipelines. I rebased to get up-to-date CI running. Let's see it pass, then we can think about redirecting to the 10.13.0 release branch.

**Comment by pako81** [comment]
CI passed. I have changed the base branch of the PR. @jvillafanez I guess this is still missing your approval. About the changelog, if we are not able to come up with an understandable one I am tempted to merge this anyway.

**Review by jvillafanez — APPROVED** [review]
I guess changes in the translation files are harmless

**Comment by phil-davis** [comment]
> I guess changes in the translation files are harmless

Note: those were already merged to the release branch yesterday by PR #40924 
If this PR is rebased, then the Transifex changes would go away. But we don't need to bother doing that.

**Comment by phil-davis** [comment]
@jvillafanez are you able to write a changelog, or give me some idea about a changelog and I can write one?

**Comment by jvillafanez** [comment]
Not sure if we need a changelog. The PR adds the `unlock` operation to the client.js, but i don't see it used here, so it shouldn't have any consequence to the customers, at least for now. I don't know if it's used somewhere else.

**Comment by phil-davis** [comment]
> Not sure if we need a changelog

OK, then we can merge - @pako81

**Pull request merged**